### DISCUSSION TITLE
[vcpkg] Support Code Runner

### DISCUSSION THREAD

**Discussion by pablanche** [microsoft/vcpkg#17830] [Ideas]
I installed the C++ boost library using vcpg. 
Now all the  squiggles marks have disappeared in my VS Code document. So the path should be OK.
However, when I run my cpp code in VS Code using Code Runner (0.9.16), I have the following message:
```
Mycode.cpp:8:11: fatal error: boost/iterator/iterator_facade.hpp: No such file or directory
    8 |  #include <boost/iterator/iterator_facade.hpp>
      |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
```
How should I configure  Code Runner (or else) so it finds the correct path?

**Comment by ghost** [comment]
@pablanche vcpkg is not designed to work with VS CODE or CODE RUNNER plugin to VS CODE.
Prerequisites:

Windows 10, 8.1, 7, Linux, or MacOS
Visual Studio 2015 Update 3 or newer (on Windows)
Git
gcc >= 7 or equivalent clang (on Linux)
Optional: CMake 3.12.4

**Comment by JackBoosY** [comment]
@strega-nil Any plain to support VS Code?

**Comment by pablanche** [comment]
Thanks for the clarification timautry. 
If I may clarify, VS CODE is more of an editor, so there should be no conflict there. CODE RUNNER is compiling and running the code through gcc (I have gcc 9.2.1). I also have Git 2.20.1. So I am still puzzled.

**Comment by ghost** [comment]
@pablanche 
cc:  @strega-nil  @JackBoosY 

vcpkg has built-in integration processes as follows:
 - on windows Visual Studio:
   - vcpkg integrate install - for system wide integration for all projects
   - vcpkg integrate project - which creates a nupkg to integrate vcpkg into project individually
 - on linux/macOS, cygwin, mingw:
   - vcpkg integrate bash - creates a script in the home directory executed to setup the environment variables needed to let gcc know where to find the vcpkg libraries and sources for the ports involved.

CODE RUNNER would need some type of configuration page or file that lets one put in the folder/directories for the vcpkg and cmake scripts, binaries, libraries, symbols and sources for all the ports vcpkg has installed. 

If there are enough requests, a **feature request** could be initiated to see what would be needed to integrate with CODE RUNNER and/or a plugin for VSCODE.

**Comment by strega-nil** [comment]
Code Runner is not currently a supported way of building with vcpkg; we may add support eventually, but more likely it should be added by the community (or, even better, vcpkg support should be added to Code Runner). However, we do have features hopefully coming down the pipeline which will make life a lot easier for community members adding that support.

**Comment by pablanche** [comment]
Thanks for your feed back. Looking forward to have vcpkg  supported.

**Comment by JackBoosY** [comment]
@pablanche Please keep this issue open before we support them.
